serialization/pickle.ksy

```diff
@@ -0,0 +1,443 @@
+            #'opcode::setitem': no_arg
+            #'opcode::setitem2': no_arg
+            #'opcode::pop': no_arg
+            #'opcode::dup': no_arg
+            #'opcode::mark': no_arg
+            #'opcode::pop_mark': no_arg
+            'opcode::get': decimalnl_short
+            'opcode::binget': u1
+            'opcode::long_binget': u4
+            'opcode::put': decimalnl_short
+            'opcode::binput': u1
+            'opcode::long_binput': u4
+            'opcode::ext1': u1
+            'opcode::ext2': u2
+            'opcode::ext4': u4
+            'opcode::global':  stringnl_noescape_pair
+            #'opcode::reduce': no_arg
+            #'opcode::build': no_arg
+            'opcode::inst': stringnl_noescape_pair
+            #'opcode::obj': no_arg
+            #'opcode::newobj': no_arg
+            'opcode::proto': u1
+            #'opcode::stop': no_arg
+            'opcode::persid': stringnl_noescape
+            #'opcode::binpersid': no_arg
+        doc: |
+          Optional argument for the operation. Data type and length
+          are determined by the value of the opcode.
+
+  decimalnl_short:
+    seq:
+      - id: val
+        type: str
+        encoding: ASCII
+        terminator: 0x0a # "\n"
+    doc: Integer, encoded with the ASCII characters [0-9-].
+  decimalnl_long:
+    seq:
+      - id: val
+        type: str
+        encoding: ASCII
+        terminator: 0x0a # "\n"
+    doc: Integer, encoded with the ASCII chracters [0-9-], followed by 'L'.
+  # TODO Can kaitai express constraint that these are quoted?
+  stringnl:
+    seq:
+      - id: val
+        terminator: 0x0a # "\n"
+    doc: Quoted string, possibly containing Python string escapes.
+  stringnl_noescape:
+    seq:
+      - id: val
+        terminator: 0x0a # "\n"
+    doc: Unquoted string, does not contain string escapes.
+  stringnl_noescape_pair:
+    seq:
+      - id: val1
+        type: stringnl_noescape
+      - id: val2
+        type: stringnl_noescape
+    doc: Pair of unquoted, unescaped strings.
+  unicodestringnl:
+    seq:
+      - id: val
+        type: str
+        encoding: ASCII
+        terminator: 0x0a # "\n"
+    doc: Unquoted string, containing Python Unicode escapes.
+  floatnl:
+    seq:
+      - id: val
+        type: str
+        encoding: ASCII
+        terminator: 0x0a # "\n"
+    doc: Double float, encoded with the ASCII characters [0-9.eE-], or 'nan'.
+  long1:
+    seq:
+      - id: len
+        type: u1
+      - id: val
+        size: len
+    doc: |
+      Large signed integer, in the range -2**(8*255-1) to 2**(8*255-1)-1,
+      encoded as two's complement.
+  long4:
+    seq:
+      - id: len
+        type: u4
+      - id: val
+        size: len
+    doc: |
+      Large signed integer, in the range -2**(8*2**32-1) to 2**(8*2**32-1)-1,
+      encoded as two's complement.
+  string1:
+    seq:
+      - id: len
+        type: u1
+      - id: val
+        type: str
+        encoding: latin1
```

Comment:
Where does this encoding come from? Is it known to always be Latin-1? I'm asking because some formats, especially older ones, appear to use Latin-1 as their string encoding, but they actually use a system-/locale-dependent default encoding (which might often be Latin-1, but could also be something else), or treat all strings as raw bytes without assuming any encoding.

Reply:
I'm investigating this, but it may be a few days the answer deserves citations

Reply:
Boy were you right to question this

Reply:
@dgelessus I hope https://github.com/kaitai-io/kaitai_struct_formats/pull/150/commits/31ea16e013c2627d3c9c80e86e428d63b34441b9#diff-b8c96cd5bf4bd59885b667523ba9466bR184 answers this to your satisfaction

Reply:
Ah, typical Python 2 encoding handling I see 😃 I agree with changing the type here from `str` to a plain byte array - when nothing is known for certain about the encoding, it makes sense to keep it as bytes and let the user of the spec deal with it.

Thank you for doing the research on this! It's good to have edge cases like this documented properly, especially when other sources are unclear on the topic.